uwblueprint/focus-on-nature
# feat: check camp status before edit

## Notion ticket link

[Checking Camp Status Before Edit](https://uwblueprintexecs.notion.site/Checking-Camp-Status-Before-Edit-0e671b345f22410e9712f91014b93488)



## Implementation description
* Throw error message if camp is active and `fee` is trying to be updated 
* Throw an error message if camp is active and requested capacity change is less than the current number of campers 
* Updated campValidators to make `fee` optional when updating the camp
* Updated campValidators to make `dates` optional when updating campSession



## Steps to test
1. Test trying to update `fee` when camp is active
* Make a PATCH request at `http://localhost:5000/camp/:campId` 
* Sample request body: 
```
{
"ageLower": 15,
"ageUpper": 30,
"campCoordinators": ["61fb3d34272ea0002ad6a24d"],
"campCounsellors": ["61fb3d34272ea0002ad6a24d"],
"name": "test camp",
"description": "description",
"earlyDropOff": "12:30",
"latePickup": "2:30",
"location": "canada",
"fee": 25,
"volunteers": ["jason"],
"startTime":"12:30",
"endTime":"2:30",
"active":true
}
```
* if the camp is active, should see `"error": "Cannot update fee of active camp"`

<br>

2. Test trying to reduce `capacity` to less than the current number of campers 
* make a PATCH request at `http://localhost:5000/camp/:campId/session/:campSessionId`
* Sample request body:
```
{
    "capacity": 1
}
```
* If there are already campers registered to the campSession and the capacity is less than the number of campers, should show error message: `"error": "Cannot decrease capacity to current number of registered campers..."`



## What should reviewers focus on?
* Are the changes to the optional fields in the validator correct? 
* Can the code be written in a more concise way? 
* Does the implementation handle all edge cases? 


## Checklist
